### ISSUE TITLE
HostTestCase.test_positive_update_puppet_class needs to create env with correct location

### ISSUE THREAD

**Issue by jhutar** [SatelliteQE/robottelo#6282]
```
2018-09-24 14:36:49 - nailgun.client - WARNING - Received HTTP 422 response: {
  "error": {"id":7606,"errors":{"environment_id":["with id 9 doesn't exist or is not assigned to proper organization and/or location"]},"full_messages":["Environment with id 9 doesn't exist or is not assigned to proper organization and/or location"]}
}
```

```
________________ HostTestCase.test_positive_update_puppet_class ________________

self = <tests.foreman.api.test_host.HostTestCase testMethod=test_positive_update_puppet_class>

    @run_only_on('sat')
    @tier1
    def test_positive_update_puppet_class(self):
        """Update a host with a new puppet classes
    
            :id: 73f9efce-3807-4196-b4e3-a6bfbfe95c99
    
            :expectedresults: A host is update with a new puppet classes
    
            :CaseImportance: Critical
            """
        host = entities.Host(organization=self.org, location=self.loc).create()
        self.assertEqual(len(host.puppetclass), 0)
        host.environment = self.env
        host.puppetclass = self.puppet_classes
>       host = host.update(['environment', 'puppetclass'])
```

I have reproduced like this after test ran locally:

```
[root@sat640snap17 ~]# hammer -u admin -p changeme host update --name ujjgkvrzuprv --environment KT_szbDIh_Library_AtBkHZD_19
Could not update the host:
  Environment with id 9 doesn't exist or is not assigned to proper organization and/or location
```

Then I have added proper location to the env and now it worked:

```
[root@sat640snap17 ~]# hammer -u admin -p changeme host update --name ujjgkvrzuprv --environment KT_szbDIh_Library_AtBkHZD_19
Host updated.
```

**Comment by ntkathole** [comment]
similar to #6272

**Comment by swadeley** [comment]
Hello, this issue was fixed by the pull request (#6324) for issue #6272.
